### PR TITLE
e2e test: unified the clients

### PR DESCRIPTION
**Pull request by riya-singhal31**
replaced all clients with a controller runtime client.

Signed-off-by: riya-singhal31 <riyasinghalji@gmail.com>

### CODE REVIEW

File: pkg/deploymanager/config.go [hunk 1]
```diff
@@ -1,133 +1,63 @@
...
-func (t *DeployManager) GetParameterCodec() runtime.ParameterCodec {
-	return t.parameterCodec
-}
-
-// GetLvmClient is the function used to retrieve the lvm client.
-func (t *DeployManager) GetLvmClient() *rest.RESTClient {
-	return t.lvmClient
-}
-
-func getKubeconfig(kubeconfig string) (*rest.Config, error) {
```
Comment (nbalacha) [inline pkg/deploymanager/config.go:51]: You can leave this function for better readibility.

File: pkg/deploymanager/config.go [hunk 2]
```diff
@@ -1,133 +1,63 @@
...
 	}
 
 	return &DeployManager{
-		olmClient:      olmClient,
-		k8sClient:      k8sClient,
-		crClient:       crClient,
-		lvmClient:      lvmClient,
-		parameterCodec: parameterCodec,
+		crClient: crClient,
+		Ctx:      context.TODO(),
```
Comment (nbalacha) [inline pkg/deploymanager/config.go:61]: Please remove the ctx from the deploymanager. Lets stick to context.TODO for now.

File: pkg/deploymanager/subscription.go [hunk 3]
```diff
@@ -97,54 +97,35 @@ func (t *DeployManager) generateClusterObjects(lvmCatalogImage string, subscript
...
-		return err
-	}
-
-	err = utilwait.PollImmediate(interval, timeout, func() (done bool, err error) {
-		pods, err := t.k8sClient.CoreV1().Pods(InstallNamespace).List(context.TODO(), metav1.ListOptions{
-			LabelSelector: labelSelector.String(),
-		})
+	err := utilwait.PollImmediate(interval, timeout, func() (done bool, err error) {
+		existingCatalogSource := &v1alpha1.CatalogSource{}
+		err = t.crClient.Get(t.Ctx, client.ObjectKeyFromObject(catalogsource), existingCatalogSource)
```
Comment (nbalacha) [inline pkg/deploymanager/subscription.go:109]: Why not just use the catalogSource name?

**Review by nbalacha — CHANGES_REQUESTED** [review]

File: pkg/deploymanager/config.go (hunk 1 above)
Comment (riya-singhal31) [reply pkg/deploymanager/config.go:51]: means to use the separate function for geting kubeconfig?

File: pkg/deploymanager/subscription.go [hunk 4]
```diff
@@ -97,54 +97,35 @@ func (t *DeployManager) generateClusterObjects(lvmCatalogImage string, subscript
...
-				if condition.Type == k8sv1.PodReady && condition.Status == k8sv1.ConditionTrue {
-					isReady = true
-					break
-				}
-			}
-		}
-
-		if !isReady {
-			lastReason = "waiting on lvm catalog source pod to reach ready state"
+		if existingCatalogSource.Status.GRPCConnectionState.LastObservedState != "READY" {
```
[the remark this answers is not in the document]
Comment (riya-singhal31) [reply pkg/deploymanager/subscription.go:114]: Yes

**Comment by riya-singhal31** [comment]
Updated the PR with all the changes.

File: pkg/deploymanager/config.go (hunk 1 above)
Comment (nbalacha) [reply pkg/deploymanager/config.go:51]: Yes

File: pkg/deploymanager/subscription.go [hunk 5]
```diff
@@ -97,7 +99,7 @@ func (t *DeployManager) generateClusterObjects(lvmCatalogImage string, subscript
 }
 
 // Waiting for LVM catalog source.
-func (t *DeployManager) waitForLVMCatalogSource() error {
+func (t *DeployManager) waitForLVMCatalogSource(catalogsource *v1alpha1.CatalogSource) error {
```
Comment (nbalacha) [inline pkg/deploymanager/subscription.go:102]: It is enough to just pass the name of the catalogsource,

File: pkg/deploymanager/config.go (hunk 1 above)
Comment (riya-singhal31) [reply pkg/deploymanager/config.go:51]: Updated that.

File: pkg/deploymanager/subscription.go (hunk 5 above)
Comment (riya-singhal31) [reply pkg/deploymanager/subscription.go]: Updated. Name is also not required now as we are following the previous method itself.

File: pkg/deploymanager/subscription.go [hunk 6]
```diff
@@ -215,36 +222,36 @@ func (t *DeployManager) deployClusterObjects(co *clusterObjects) error {
...
 			return err
 		}
-	}
 
-	// Wait for catalog source before posting subscription
-	err := t.waitForLVMCatalogSource()
-	if err != nil {
-		return err
-	}
+		// Wait for catalog source before posting subscription
```
[the remark this answers is not in the document]
Comment (riya-singhal31) [reply pkg/deploymanager/subscription.go:238]: I was passing catalog source as an argument to the waitForLVMCatalogSource call, so thats why did that. But now its of no use.
Shall I move it outside?

[the remark this answers is not in the document]
Comment (riya-singhal31) [reply pkg/deploymanager/subscription.go]: Updated, thanks.

**Review by nbalacha — APPROVED** [review]

**Pull request merged**